docs/spec/openapi/public-docs.yaml

```diff
@@ -0,0 +1,5703 @@
+                    description: >-
+                      The maximum number of spans that are recorded per
+                      transaction.
+                    example: 500
+                  transaction_sample_rate:
+                    type: number
+                    format: float
+                    minimum: 0
+                    maximum: 1
+                    description: The agent samples transactions at this rate.
+                    example: 0.3
+                additionalProperties: true
+        '403':
+          description: >-
+            APM Server is configured to fetch agent configuration from
+            Elasticsearch but the configuration is invalid.
+        '503':
+          description: APM Server is starting up or Elasticsearch is unreachable.
+  /config/v1/rum/agents:
+    get:
+      summary: Get RUM agent configuration changes
+      description: >
+        To configure authenticated access to the APM server, the instructions at
+        [APM API
+        key](https://www.elastic.co/guide/en/observability/current/api-key.html)
+        or [APM Secret
+        Token](https://www.elastic.co/guide/en/observability/current/secret-token.html)
+        must be followed to configure the correct permissions for APM access.
+      operationId: getRumAgentConfig
+      tags:
+        - agent config
+      responses:
+        '200':
+          description: Successful response.
+  /intake/v2/events:
+    post:
+      summary: Send APM agent events
+      description: >
+        NOTE: Most users do not need to interact directly with the events intake
+        API. Agents communicate with the APM Server by sending events--captured
+        pieces of information--in an HTTP request. Events can be transactions,
+        spans, errors, or metrics. Each event is sent as its own line in the
+        HTTP request body, which is known as newline delimited JSON (NDJSON).
+        With NDJSON, agents can open an HTTP POST request and use chunked
+        encoding to stream events to the APM Server as soon as they are recorded
+        in the agent. This makes it simple for agents to serialize each event to
+        a stream of newline delimited JSON. The APM Server also treats the HTTP
+        body as a compressed stream and thus reads and handles each event
+        independently. See the [APM data
+        model](https://www.elastic.co/guide/en/observability/current/apm-data-model.html)
+        to learn more about the different types of events.
+      operationId: postEventIntake
+      tags:
+        - event intake
+      requestBody:
+        content:
+          application/ndjson:
+            schema:
+              anyOf:
+                - $ref: '#/components/schemas/ErrorEvent'
+                - $ref: '#/components/schemas/MetadataEvent'
+                - $ref: '#/components/schemas/MetricSetEvent'
+                - $ref: '#/components/schemas/SpanEvent'
+                - $ref: '#/components/schemas/TransactionEvent'
```

Comment:
Every requests needs to have the `MetadataEvent` as the first line in the `ndjson` request, see https://www.elastic.co/docs/solutions/observability/apm/elastic-apm-events-intake-api#apm-api-metadata-overview.